integ-test-promote.pipeline.yml

```diff
@@ -1,485 +1,563 @@
-                    pipeline: 6
-                    runVersion: latest
-                    artifact: extension
-                    path: $(Build.StagingDirectory)
-
-                - task: TfxInstaller@2
-                  displayName: Install tfx-cli
-                  inputs:
-                    version: v0.7.x
-
-                - task: PublishAzureDevOpsExtension@2
-                  displayName: Publish release
-                  inputs:
-                    connectTo: VsTeam
-                    connectedServiceName: MathWorks
-                    fileType: vsix
-                    vsixFile: $(Build.StagingDirectory)/*.vsix
-                    updateTasksVersion: false
+resources:
+  pipelines:
+    - pipeline: integ_test_promote
+      source: pre-integ-test
+      trigger: true
+
+trigger: none
+pr: none
+
+stages:
+  - stage: integration_test
+    displayName: Integration test
+    variables:
+      - group: matlab-batch-token
+    jobs:
+      - job: test_install_v0
+        strategy:
+          matrix:
+            microsoft_hosted_linux:
+              poolName: Azure Pipelines
+              vmImage: ubuntu-22.04
+            microsoft_hosted_macos:
+              poolName: Azure Pipelines
+              vmImage: macOS-latest
+            microsoft_hosted_windows:
+              poolName: Azure Pipelines
+              vmImage: windows-latest
+        pool:
+          name: $(poolName)
+          vmImage: $(vmImage)
+        steps:
+          - checkout: none
+          - task: MathWorks.matlab-azure-devops-extension-dev.InstallMATLAB.InstallMATLAB@0
+            displayName: Install MATLAB
+          - bash: |
+              set -e
+              matlab -batch version
+              os=$(uname)
+              if [[ $os = CYGWIN* || $os = MINGW* || $os = MSYS* ]]; then
+                mex.bat -h
+              else
+                mex -h
+              fi
+            displayName: Check matlab and mex
+
+      - job: test_install_release_v0
+        strategy:
+          matrix:
+            microsoft_hosted_linux:
+              poolName: Azure Pipelines
+              vmImage: ubuntu-22.04
+            microsoft_hosted_macos:
+              poolName: Azure Pipelines
+              vmImage: macOS-latest
+            microsoft_hosted_windows:
+              poolName: Azure Pipelines
+              vmImage: windows-latest
+        pool:
+          name: $(poolName)
+          vmImage: $(vmImage)
+        steps:
+          - checkout: none
+          - task: MathWorks.matlab-azure-devops-extension-dev.InstallMATLAB.InstallMATLAB@0
+            displayName: Install MATLAB
+            inputs:
+              release: R2021a
+          - bash: |
+              set -e
+              matlab -batch "assert(strcmp(version('-release'),'2021a'))"
+            displayName: Check matlab release
+            env:
+              MATHWORKS_ACCOUNT: $(MATHWORKS_ACCOUNT)
+              MATHWORKS_TOKEN: $(MATHWORKS_TOKEN)

```

Comment:
Not needed?

Reply:
If I understood the Azure docs correctly, the variables get loaded for use in the yaml for pipeline config from the lines
```
    variables:
      - group: matlab-batch-token
``` 
but it doesn't get added to the agent's environment  unless you do this

Reply:
But I wouldn't expect v0 to need these env vars even when calling "matlab -batch" directly.

Reply:
Oh sorry, I didn't realize this was on the v0 test here as well. Thanks for catching this